### PR TITLE
Ensure `BatcherConfidential` intermediate steps do not transfer underlying `toToken`

### PR DESCRIPTION
**Pull request by arr00**
## Summary by CodeRabbit

* **Bug Fixes**
  * BatcherConfidential now reverts when underlying tokens are received during partial route execution, strengthening contract safety and preventing invalid states.

* **Tests**
  * Added comprehensive test coverage for partial route execution scenarios, including multi-step routes and token transfer validation edge cases.

### CODE REVIEW

File: .changeset/tasty-badgers-walk.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'openzeppelin-confidential-contracts': minor
+---
+
+`BatcherConfidential`: revert if underlying to token balance changes during a partial route execution.
```
Comment (arr00) [inline .changeset/tasty-badgers-walk.md]:
```suggestion
`BatcherConfidential`: revert if underlying `toToken` balance changes during a partial route execution.
```

**Review by Amxx — DISMISSED** [review]
